src/v/cluster/self_test/cloudcheck.cc

```diff
@@ -226,12 +235,11 @@ ss::future<self_test_result> cloudcheck::verify_upload(
         co_return result;
     }
 
-    auto rtc = retry_chain_node(_opts.timeout, _opts.backoff, &_rtc);
-    cloud_storage::upload_request upload_request = make_upload_request(
-      bucket, key, payload.value().copy(), rtc);
-
     const auto start = ss::lowres_clock::now();
     try {
+        auto rtc = retry_chain_node(_opts.timeout, _opts.backoff, &_rtc);
+        cloud_storage::upload_request upload_request = make_upload_request(
+          bucket, key, payload.value().copy(), rtc);
```

Comment:
will this have the side effect of exposing potentially internal errors as test result messages and be confusing? maybe it's not worth trying to optimize it, but just a thought.

Reply:
Good point. I don't expect to see the construction of either of these objects throwing or exposing any internal errors, however.

This change of moving construction of the `rtc` into the `try` block is primarily for better scoping of resources, not error-handling.